### PR TITLE
[Refactor] Refactor BaseTableInfo to be more clear

### PR DESCRIPTION
**Pull request by LiShuMing**
Why I'm doing:

`BaseTableInfo ` is meta data and should not contain much util functions.

What I'm doing:
- Remove some redundant methods from `BaseTableInfo` and add more checks.

Fixes #issue

## What type of PR is this:

- [ ] BugFix
- [ ] Feature
- [ ] Enhancement
- [x] Refactor
- [ ] UT
- [ ] Doc
- [ ] Tool

Does this PR entail a change in behavior?

- [ ] Yes, this PR will result in a change in behavior.
- [x] No, this PR will not result in a change in behavior.

If yes, please specify the type of change:

- [ ] Interface/UI changes: syntax, type conversion, expression evaluation, display information
- [ ] Parameter changes: default values, similar parameters but with different default values
- [ ] Policy changes: use new policy to replace old one, functionality automatically enabled
- [ ] Feature removed
- [ ] Miscellaneous: upgrade & downgrade compatibility, etc.

## Checklist:

- [x] I have added test cases for my bug fix or my new feature
- [ ] This pr needs user documentation (for new or modified features or behaviors)
  - [ ] I have added documentation for my new feature or new function
- [ ] This is a backport pr

## Bugfix cherry-pick branch check:
- [x] I have checked the version labels which the pr will be auto-backported to the target branch
  - [ ] 3.2
  - [ ] 3.1
  - [ ] 3.0
  - [ ] 2.5

### CODE REVIEW

**Review by murphyatwork — DISMISSED** [review]

File: fe/fe-core/src/main/java/com/starrocks/catalog/BaseTableInfo.java [hunk 1]
```diff
@@ -162,92 +179,62 @@ public Table getTableChecked() {
...
 
+    /**
+     * Get {@link Table} from {@code BaseTableInfo} which it can be an OlapTable or ExternalTable.
+     * </p>
+     * NOTE: Prefer {@link BaseTableInfo#getTableChecked()} or {@link BaseTableInfo#mayGetTable()}
+     * because this method may return null if base table has already dropped or schema changed.
+     * </p>
+     * @return {@link Table} if BaseTableInfo is health otherwise null will be returned.
+     */
     @Deprecated
```
Comment (nshangyiming) [inline fe/fe-core/src/main/java/com/starrocks/catalog/BaseTableInfo.java:190]: when to make this api private?

**Review by nshangyiming — DISMISSED** [review]

File: fe/fe-core/src/main/java/com/starrocks/catalog/BaseTableInfo.java (hunk 1 above)
Comment (LiShuMing) [reply fe/fe-core/src/main/java/com/starrocks/catalog/BaseTableInfo.java:190]: later... because this will change too much codes.

**Review by murphyatwork — APPROVED** [review]

File: fe/fe-core/src/main/java/com/starrocks/catalog/ForeignKeyConstraint.java [hunk 2]
```diff
@@ -165,12 +167,23 @@ public static List<ForeignKeyConstraint> parse(String foreignKeyConstraintDescs)
...
+            Preconditions.checkArgument(NumberUtils.isNumber(table),
+                    "BaseTableInfo table %s should be tableId for internal catalog", table);
+            long dbId = Long.parseLong(db);
+            long tableId = Long.parseLong(table);
+            Database database = GlobalStateMgr.getCurrentState().getDb(dbId);
+            Preconditions.checkArgument(database != null, "BaseInfo's db %s should not null", dbId);
+            Table baseTable = database.getTable(tableId);
+            Preconditions.checkArgument(baseTable != null,
+                    "BaseInfo's baseTable %s should not null", tableId);
+            baseTableInfo = new BaseTableInfo(dbId, database.getFullName(), baseTable.getName(), tableId);
```
Comment (ABingHuang) [inline fe/fe-core/src/main/java/com/starrocks/catalog/ForeignKeyConstraint.java:184]: what will happen if these checks throw exception?

Comment (LiShuMing) [reply fe/fe-core/src/main/java/com/starrocks/catalog/ForeignKeyConstraint.java:184]: ignore the constraints.
```

    public TableProperty buildConstraint() {
        if (properties.containsKey(PropertyAnalyzer.PROPERTIES_UNIQUE_CONSTRAINT)) {
            try {
                uniqueConstraints = UniqueConstraint.parse(
                        properties.getOrDefault(PropertyAnalyzer.PROPERTIES_UNIQUE_CONSTRAINT, ""));
            } catch (Exception e) {
                LOG.warn("Failed to parse unique constraints, ignore this unique constraint", e);
            }
        }

        if (properties.containsKey(PropertyAnalyzer.PROPERTIES_FOREIGN_KEY_CONSTRAINT)) {
            try {
                foreignKeyConstraints = ForeignKeyConstraint.parse(
                        properties.getOrDefault(PropertyAnalyzer.PROPERTIES_FOREIGN_KEY_CONSTRAINT, ""));
            } catch (Exception e) {
                LOG.warn("Failed to parse foreign key constraints, ignore this foreign key constraints", e);
            }
        }
        return this;
    }
```

**Review by ABingHuang — APPROVED** [review]

**Review by Seaven — APPROVED** [review]

**Review by nshangyiming — APPROVED** [review]

**Pull request merged**